haskell/cabal.bzl

```diff
@@ -1556,6 +1916,16 @@ def stack_snapshot(
     `<package>-<version>` in the `packages` attribute. Note that you cannot
     override the version of any [packages built into GHC][ghc-builtins].
 
+    This rule invokes the `stack` tool for version and dependency resolution
+    based on the specified snapshot. You can generate a `stack_snapshot.json`
+    file to avoid invoking `stack` on every fetch and instead pin the outcome
+    in a file that can be checked into revision control. Execute the following
+    command and then specify the `stack_snapshot_json` attribute:
+
+    ```
+    bazel run @stackage-unpinned//:pin
+    ```
```

Comment:
We should add some docs on how to update the snapshot.

Reply:
Ah, it’s in the pin script, but mentioning it here as well can’t hurt.

Reply:
I've expanded the docs a bit.